src/renderer/extensions/vueNodes/composables/useSlotLinkInteraction.ts

```diff
@@ -359,18 +486,36 @@ export function useSlotLinkInteraction({
     if (!pointerSession.matches(event)) return
     event.preventDefault()
 
+    raf.flush()
+
     if (!state.source) {
       cleanupInteraction()
       app.canvas?.setDirty(true)
       return
     }
 
-    const candidate = candidateFromTarget(event.target)
-    let connected = tryConnectToCandidate(candidate)
+    // Prefer using the snapped candidate captured during hover for perf + consistency
+    const snappedCandidate = state.candidate?.compatible
+      ? state.candidate
+      : null
+
+    let connected = tryConnectToCandidate(snappedCandidate)
+
+    // Fallback to DOM slot under pointer (if any), then node fallback, then reroute
+    if (!connected) {
+      const domCandidate = candidateFromTarget(event.target)
+      connected = tryConnectToCandidate(domCandidate)
+    }
+
+    if (!connected) {
+      const nodeCandidate = candidateFromNodeTarget(event.target)
+      connected = tryConnectToCandidate(nodeCandidate)
+    }
+
     if (!connected) connected = tryConnectViaRerouteAtPointer() || connected
```

Comment:
nit: This kind of long chain of optional assignment to a mutable variable is a great case for extraction to a more pure function.